app/src/main/java/me/ccrama/redditslide/Activities/Settings.java

```diff
@@ -33,33 +53,128 @@
+    public static boolean                                            changed;  //whether or not a Setting was changed
+
+    private SettingsGeneralFragment      mSettingsGeneralFragment      = new SettingsGeneralFragment(this);
+    private ManageOfflineContentFragment mManageOfflineContentFragment = new ManageOfflineContentFragment(this);
+    private SettingsThemeFragment        mSettingsThemeFragment        = new SettingsThemeFragment(this);
+    private SettingsFontFragment         mSettingsFontFragment         = new SettingsFontFragment(this);
+    private SettingsCommentsFragment     mSettingsCommentsFragment     = new SettingsCommentsFragment(this);
+    private SettingsHandlingFragment     mSettingsHandlingFragment     = new SettingsHandlingFragment(this);
+    private SettingsHistoryFragment      mSettingsHistoryFragment      = new SettingsHistoryFragment(this);
+    private SettingsDataFragment         mSettingsDataFragment         = new SettingsDataFragment(this);
+    private SettingsRedditFragment       mSettingsRedditFragment       = new SettingsRedditFragment(this);
 
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (requestCode == RESTART_SETTINGS_RESULT) {
-            Intent i = new Intent(Settings.this, Settings.class);
-            i.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
-            i.putExtra("position", scrollY);
-            startActivity(i);
-            overridePendingTransition(0, 0);
-            finish();
-            overridePendingTransition(0, 0);
+            restartActivity();
         }
     }
 
+    public void restartActivity() {
+        Intent i = new Intent(Settings.this, Settings.class);
+        i.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
+        i.putExtra("position", scrollY);
+        i.putExtra("prev_text", prev_text);
+        startActivity(i);
+        overridePendingTransition(0, 0);
+        finish();
+        overridePendingTransition(0, 0);
+    }
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         applyColorTheme();
         setContentView(R.layout.activity_settings);
         setupAppBar(R.id.toolbar, R.string.title_settings, true, true);
 
+        if (getIntent() != null && !Strings.isNullOrEmpty(getIntent().getStringExtra("prev_text"))) {
+            prev_text = getIntent().getStringExtra("prev_text");
+        } else if (savedInstanceState != null) {
+            prev_text = savedInstanceState.getString("prev_text");
+        }
+
+        if (!Strings.isNullOrEmpty(prev_text)) {
+            ((EditText) findViewById(R.id.settings_search)).setText(prev_text);
+        }
+
+        BuildLayout(prev_text);
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        outState.putInt("position", scrollY);
+        outState.putString("prev_text", prev_text);
+        super.onSaveInstanceState(outState);
+    }
+
```

Comment:
Think it would be worth it here now or in the future to just loop over all the available layouts we want to deal with and run addView for each item? It'd just be a static array of Integers anyway.

Reply:
You mean once during initialization? Currently I remove basically everything (that may already have been pruned down), re-add it, then prune it again every time the search text input changes.

Reply:
I meant more for code simplification / clarity than anything. This method just called addView for every view we want added so it chains and compounds as more things are potentially added.

Reply:
Oh I see, just do it iteratively so that all that would change are the array contents.

Reply:
Done